### DISCUSSION TITLE
Performance: Reanimated 2 vs. Animated

### DISCUSSION THREAD

**Discussion by dbi1** [software-mansion/react-native-reanimated#2860] [General]
Hello,

I'm trying to understand what the performance difference (if any) between Reanimated 2 and React Native's own Animated library is.

My understanding is that both use the UI thread to run the animation, so the running itself should be equal.

Might there be a difference in how fast the animation is started?

Thank you kindly for any thoughts.

P.S.: I understand that Reanimated 2 has many advantages over Animated, but here I'm simply interested in the performance aspect.

**Comment by JasonCareter** [comment]
All animation of Reanimated are defined on js and run on ui thread . The animation of RN's Animated Library can also run on ui thread with useNativeDriver: true . The first critical difference is that there is no synchronous way to get the current animation value . With reanimated ,you can do that . By the way , reanimated is based on RN's new architecture TurboModule ( jsi ) . which means , js code can communicate with C++ directly without bridge . It saved a lot of performance on convert data and so on . There are some properties cannot move to native on RN's Animated Library  , like width ,height and opacity . You can do that with reanimated.
